python/nano/src/bigdl/nano/pytorch/inference/pipeline.py

```diff
@@ -91,10 +97,23 @@ def run(self, inputs):
     def _launch_stage(self, stage: Tuple, recv_: Connection, send_: Connection):
         name, func, config = stage
 
-        p = mp.Process(target=self._stage_wrapper,
-                       args=(func, recv_, send_),
-                       daemon=True)
-        p.start()
+        subprocess_env = {}
+        if isinstance(config.get("core_num", None), int):
+            core_num = config["core_num"]
+            cores = self.cores[:core_num]
+            self.cores = self.cores[core_num:]
+            if len(cores) < core_num:
+                warnings.warn(f"stage {name} requires {core_num} cores,"
+                              f" but there are only {len(cores)} cores left")
+            subprocess_env["KMP_AFFINITY"] = (f"granularity=fine,proclist"
+                                              f"=[{','.join([str(i) for i in cores])}],explicit")
+            subprocess_env["OMP_NUM_THREADS"] = str(len(cores))
```

Comment:
I think we may just allow users to state the environment variables themselves. `core_num` seems to be a good shortcut for users but shortcut always has some limitations (e.g., this `core_num` only controls the openmp workload's thread num with a specific affinity strategy)

Reply:
We may provide another key named `env` for users to override the environment variables of subprocess ? e.g.
`pipeline = Pipeline([("preprocess", preprocess, {"core_num": 4, "env": {"KMP_AFFINITY": "granularity=fine,none"}}])`